### DISCUSSION TITLE
Testing Joyride using React-Testing-Library?

### DISCUSSION THREAD

**Discussion by astrosuf** [gilbarbara/react-joyride#821] [Q&A] [answered]
Thank you for this awesome library, I'm using react-joyride in a project of mine. We're unit-testing the application using RTL. I would like to introduce unit-tests to ensure that in a specific state the react-joyride popups are visible - however when I render the component with RTL I get an error:

`Cannot read properties of null (reading 'isConnected')`

digging a bit deeper into the error, it seems to be occurring inside react-floater.js in node_modules - the console also starts to print the lib file. 

Question: Is there some specific setup that needs to happen to test Joyride with RTL?

**Comment by gilbarbara** [comment, marked as answer]
hey @astrosuf 

This library needs a browser environment to work properly, and I don't think the JSDOM environment inside jest will do.
Anyway, you will need to change jest's [testEnvironment](https://jestjs.io/docs/configuration#testenvironment-string) to `dom` and probably mock some functionality.

Anyway, react-floater uses RTL for its tests, so it does work.
